store/src/state.rs

```diff
@@ -596,28 +625,29 @@ async fn load_nullifier_tree(db: &mut Db) -> Result<TieredSmt> {
 }
 
 #[instrument(target = "miden-store", skip_all)]
-async fn load_mmr(db: &mut Db) -> Result<Mmr> {
+async fn load_mmr(db: &mut Db) -> Result<Mmr, StateInitializationError> {
     let block_hashes: Result<Vec<RpoDigest>, ParseError> = db
         .select_block_headers()
         .await?
         .into_iter()
         .map(|b| b.try_into().map(|b: BlockHeader| b.hash()))
         .collect();
 
-    let mmr: Mmr = block_hashes?.into();
+    let mmr: Mmr = block_hashes.map_err(ConversionError::ParseError)?.into();
```

Comment:
I think we should do something similar to what we did for `load_accounts` and `load_nullifier_tree` here. Specifically, use `StateInitializationError::FailedToCreateChainMmr` variant.